File: python/calculadora.py
print('Calculadora')
operation = input('Digita la operación que quieres realizar (+. -. *, /,**)->')     # se pide a usuario elegir operacion a realizar
while operation != '+' or operation != '-' or operation != '*' or operation != '/' or operation != '**': # bucle operacion
    if operation == '+' or operation == '-' or operation == '*' or operation == '/' or operation == '**':
        num1 = input('Ingresa el primer numero ')   # hasta que no digite bien simbolo de operacion no pide 1er numero
        if num1.isdigit():
            num1 = float(num1)      # si digita algun caracter que no sea digito, se devuelve a escoger operacion
            num2 = input('Ingresa el segundo numero ')     # si digita bien primer numero, pide el segundo numero
            if num2.isdigit():
                num2 = float(num2)      # si digita bien segundo numero entra a realizar la operacion, si no, devuelve a escoger operacion
                if operation == '+':
                    add = float(num1+num2)      # operacion de suma
                    print(f'La suma de {num1} + {num2} es = {add}')
                elif operation == '-':
                    sub = float(num1-num2)      # operacion de resta
                    print(f'La resta de {num1} - {num2} es = {sub}')
                elif operation == '*':
                    mul = float(num1*num2)      # operacion de multiplicacion
                    print(f'La multiplicación de {num1} x {num2} es = {mul}')
                elif operation == '**':
                    pwr = float(num1**num2)     # operacion de potencia
                    print(f'La potencia de {num1}^{num2} es = {pwr}')
                elif operation == '/':           # operacion de division
                    if num2 == 0:               # se especifica que no se puede dividir por 0
                        print('Lo sentimos, no se puede dividir por 0')
                    else:
                        div = float(num1/num2)
                        print(f'La división de {num1} / {num2} es = {div}')
    operation = input('Digita la operación que quieres realizar (+. -. *, /,**): o "fin" para terminar el programa-> ')  # bucle operacion
    if operation == 'fin' or operation == 'FIN':      # para cerrar programa cuando el usuario decida
        break

File: java/Calculadora.java
import java.util.Scanner;

public class Calculadora {
    public static double sumar(double a, double b){
        return (a + b);
    }
    public static double restar(double a, double b){
        return (a - b);
    }
    public static double multiplicar(double a, double b){
        return (a * b);
    }
    public static Object dividir(double a, double b){
        return (b == 0) ? "No se puede dividir por 0" : (a / b);
    }
    public static double numeros(){
        Scanner teclado = new Scanner(System.in);
        System.out.println("Ingrese numero:");
        return teclado.nextDouble();
    }
    public static void main(String[] args) {
        var teclado = new Scanner(System.in);
        System.out.println("\nMenu calculadora\n1. Sumar\n2. Restar\n3. Multiplicar\n4. Dividir\n5. Salir");
        var opcion = teclado.nextInt();
        while (opcion > 0 && opcion <=5){
            if (opcion == 5){
                System.out.println("Gracias");
                Runtime.getRuntime().exit(0);
            }
            double a = numeros();
            double b = numeros();
            switch (opcion) {
                case 1:
                System.out.println("La suma de " + a + " + " + b + " es: " + sumar(a, b));
                    break;
                case 2:
                System.out.println("La resta de " + a + " - " + b + " es: " + restar(a, b));
                    break;
                case 3:
                System.out.println("La multiplicacion de " + a + " * " + b + " es: " + multiplicar(a, b));
                    break;
                case 4:
                System.out.println("La division de " + a + " / " + b + " es: " + dividir(a, b));
                    break;
                }
            System.out.println("\nMenu calculadora\n1. Sumar\n2. Restar\n3. Multiplicar\n4. Dividir\n5. Salir");
            opcion = teclado.nextInt();
            }
        teclado.close();
        }
    }